bindings/python/iota_sdk/types/block/wrapper.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2023 IOTA Stiftung
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import Dict
+from dacite import from_dict
+from iota_sdk.types.block.block import Block
+from iota_sdk.types.common import HexStr, json
+from iota_sdk.types.signature import Signature
+from iota_sdk.utils import Utils
+
+
+@json
+@dataclass
+class BlockWrapper:
+    """The two block types introduced in this TIP share some common fields which are contained in an outer wrapper.
+    This Block Wrapper is only introduced as a way to deduplicate the definition of those fields. It is not a standalone block type.
+    All fields of a Block Wrapper except for the Block and Signature constitute the Block Header.
+    """
+
+    def __init__(self, header: BlockHeader, block: Block,
+                 signature: Signature):
+        if not isinstance(block, Block):
+            raise ValueError(
+                "block must be an instance of Block or its subclasses")
+        self.header = header
+        self.block = block
+        self.signature = signature
+
+    def id(self) -> HexStr:
+        """Returns the block ID as a hexadecimal string.
+        """
+        return Utils.block_id(self)
+
+    @classmethod
+    def from_dict(cls, block_wrapper_dict: Dict) -> BlockWrapper:
+        """
+        The function `from_dict` takes a dictionary that contains the data needed to
+        create an instance of the `BlockWrapper` class.
+
+        Returns:
+
+        An instance of the `BlockWrapper` class.
+        """
+        return from_dict(BlockWrapper, block_wrapper_dict)
```

Comment:
This works with the fields just defined in the __init__ method?

Reply:
good question ... I'll investigate ...

Reply:
I don't think so as this is a class method, and the fields in the init method are instance fields.

Reply:
should be fixed now

Reply:
Hm you added it to the from_dict method, why not define the fields like we do for all the other types?
https://github.com/iotaledger/iota-sdk/blob/f289a22c27834472591111d266826bfae8738ea4/bindings/python/iota_sdk/types/output.py#L91-L100
No custom init method needed at all?

Reply:
because I got confused with the linter complaining about the `payload` field not available on block.

Reply:
the test passes so it works now